packages/application-components/src/components/modal-pages/custom-form-modal-page/custom-form-modal-page.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import { MessageDescriptor } from 'react-intl';
+import buttonMessages from '../../../utils/button-messages';
+import ModalPage from '../internals/modal-page';
+import ModalPageHeader from '../internals/modal-page-header';
+import { ContentWrapper } from '../internals/modal-page.styles';
+import { Spacings } from '@commercetools-frontend/ui-kit';
+import {
+  FormPrimaryButton,
+  FormSecondaryButton,
+  FormDeleteButton,
+} from '../../../utils/default-form-buttons';
+
+type Label = string | MessageDescriptor;
+type Props = {
+  level?: number;
+  title: string;
+  isOpen: boolean;
+  onClose?: (event: React.SyntheticEvent) => void;
+  children: React.ReactNode;
+  zIndex?: number;
+  baseZIndex?: number;
+  getParentSelector?: () => HTMLElement;
+  shouldDelayOnClose?: boolean;
+  // TopBar Props
+  topBarCurrentPathLabel?: string;
+  topBarPreviousPathLabel?: Label;
+  // Header Props
+  subtitle?: string | React.ReactElement;
+  // Controls Props
+  controls: React.ReactNode;
+  hideControls?: boolean;
+};
+
+const CustomFormModalPage = (props: Props) => (
+  <ModalPage
+    level={props.level}
+    title={props.title}
+    isOpen={props.isOpen}
+    zIndex={props.zIndex}
+    onClose={props.onClose}
+    baseZIndex={props.baseZIndex}
+    currentPathLabel={props.topBarCurrentPathLabel || props.title}
+    previousPathLabel={props.topBarPreviousPathLabel}
+    getParentSelector={props.getParentSelector}
+    shouldDelayOnClose={props.shouldDelayOnClose}
+  >
+    <ModalPageHeader title={props.title} subtitle={props.subtitle}>
+      {!props.hideControls && props.controls && (
+        <Spacings.Inline alignItems="flex-end">
```

Comment:
I'm not sure if we should leave this to the consumer 🤔

Reply:
What do you mean? Should we move the `controls` and `hideControls` to be part of the ModalPageHeader?

Reply:
I mean the Spacing wrapper. I think it can be up to the consumer to define that, I fear that we might run into some unwanted constraints by having the spacing component on our side.

Reply:
Well, in all the migrations I did in the MC, they always respect this layout.
If we need to add more flexibility in the future, might as well just add a prop to render a customHeader or a customElement in the header.
This `formControls` prop already allows rendering anything, and the only "controls specific" thing it's doing right now is placing the rendered node inside that `Spacings`.

So.. I'd prefer to keep this here 😄